framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/ProjectModification/AngularPwaSupportAdder.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using Volo.Abp.Cli.Utils;
+using Volo.Abp.DependencyInjection;
+
+namespace Volo.Abp.Cli.ProjectModification;
+
+public class AngularPwaSupportAdder : ITransientDependency
+{
+    protected ICmdHelper CmdHelper { get; }
+    protected PackageJsonFileFinder PackageJsonFileFinder { get; }
+
+    public AngularPwaSupportAdder(
+        ICmdHelper cmdHelper,
+        PackageJsonFileFinder packageJsonFileFinder)
+    {
+        CmdHelper = cmdHelper;
+        PackageJsonFileFinder = packageJsonFileFinder;
+    }
+
+    public virtual void AddPwaSupport(string rootDirectory)
+    {
+        var fileList = PackageJsonFileFinder.Find(rootDirectory).Where(x => x.Contains("angular")).ToList();
```

Comment:
It's better to check if it is next to a `angular.json` file.

PackageJsonFileFinder.Find(rootDirectory).Where(x => File.Exist( x.RemovePostFix("package.json") + "angular.json") ).ToList();

Reply:
it's folder, So I added to `Path.DirectorySeparatorChar` to the end of it

Reply:
No, you misunderstood me. You need to check if there is a `angular.json` file in the same folder, to understand if it is an angular project or not.

![image](https://user-images.githubusercontent.com/5793291/163957993-0358f61b-8354-4a1b-a493-fde931f69366.png)

Reply:
Ok got it 👍

Reply:
You can check commit https://github.com/abpframework/abp/commit/d30082bc9616c2f61aa7319aa9646a86401b79ad

Reply:
I see,

I've tested now, this way works well 👍 
https://github.com/abpframework/abp/commit/e6e5cb05a8aeb6222933f0977ef25e4cf8070399